### DISCUSSION TITLE
Digital/2024/lab1/verilog2/

### DISCUSSION THREAD

**Discussion by giscus** [wintermelon008/USTC-DigitalLab-Discussion#35] [Announcements]
# Digital/2024/lab1/verilog2/

https://soc.ustc.edu.cn/Digital/2024/lab1/verilog2/

**Comment by zucanlv** [comment]
看不懂***复位信号***

**Reply by wintermelon008** [reply]
简单一点的话只需要理解：复位信号向系统发出了一个消息，数字系统接收到这个消息后会把相应的寄存器设置成初始状态。因此这个信号和时钟信号一样，是不同于其他我们自己定义的信号的

**Comment by Lifetimer** [comment]
你好！
文章中说Xilinx建议使用同步复位的方式，那么一般建议使用高电平复位还是低电平复位？这两种同步复位的方式哪一种更适应Xilinx FPGA的结构还是都没有关系？
参考资料2中说明“如果复位信号在时钟沿附近则可能导致复位信号的亚稳态”是同步复位方式的弊端，在平常的设计中需要考虑这个弊端吗？

**Reply by wintermelon008** [reply]
这个主要看系统的实际需求。比如物理开发板上的复位按钮按下后变为低电平，无操作时保持高电平，这个时候一般用低电平复位；而FPGAOL上的复位按钮按下后变为高电平，因此一般用高电平复位。

对于精细度要求非常高的设计来说是需要考虑时钟边沿的信号突变的，也就是建立与保持时间。但我们平时的设计不会有如此高的要求，且电路规模不是很大，因此可以不考虑~

**Reply by Lifetimer** [reply]
谢谢你！
